pkg/api/timestamp.go

```diff
@@ -62,15 +82,35 @@ func TimestampResponseHandler(params timestamp.GetTimestampResponseParams) middl
 	}
 
 	// Create response
-	ctx := params.HTTPRequest.Context()
+	httpReq := params.HTTPRequest
+	ctx := httpReq.Context()
 	resp, err := RequestFromRekor(ctx, *req)
 	if err != nil {
 		return handleRekorAPIError(params, http.StatusInternalServerError, err, failedToGenerateTimestampResponse)
 	}
 
-	// TODO: Upload to transparency log and add entry UUID to location header.
-	log.Logger.Errorf("generated OK")
-	return timestamp.NewGetTimestampResponseOK().WithPayload(ioutil.NopCloser(bytes.NewReader(resp)))
+	// Upload to transparency log and add entry UUID to location header.
+	paramReq := *httpReq
+	paramReq.URL.Path = "/api/v1/log/entries"
+	entryParams := entries.CreateLogEntryParams{
+		HTTPRequest:   &paramReq,
+		ProposedEntry: createRFC3161FromBytes(resp),
+	}
+
+	// If middleware is returned, this indicates an error.
+	logEntry, middleware := createLogEntry(entryParams)
+	if middleware != nil {
+		return middleware
```

Comment:
I think this should be fine (returning an error responder from CreateLogEntry) but curious if you have artificially forced an error path here? (they both implement the same interface, so it should be fine... but since that is all part of the generated code from go-openapi I'm not 100% sure they aren't doing something funky).

Reply:
Hmmm yeah, I really didn't understand too well how repurposing the generated code would work. What do you mean by "forced an error path"? 
The error path shouldn't hit if the log entry was created. In some cases an error could legitimately happen (some internal server error). 
The other option is creating our own error struct and change the interface a little bit so that `handlRekorAPIError` can take that input

Reply:
Sorry I was unintentionally obtuse. "forced an error path" meant "temporarily change line 102 to `middleware == nil`" which would cause the rare behavior to always occur. If this causes the server to panic, then you'd know that while the compiler is happy, the generated code somewhere really isn't and we need to investigate a different approach.

Reply:
AH! That's a great idea, I was looking through what errors I could generate easily earlier to test the path, forcing it is great. Thanks!

Reply:
Seemed to go through fine, I forced a signing error to happen, returning non-nil middleware, and I ended up with
```
2021/06/22 13:14:19 [POST /api/v1/timestamp][500] getTimestampResponse default  &{Code:500 Message:Error signing}
``` 
server-side looks fine as well

Reply:
ok, then this part LGTM then.